codegen/smithy-kotlin-codegen/src/main/kotlin/software/amazon/smithy/kotlin/codegen/model/knowledge/EndpointParameterIndex.kt

```diff
@@ -44,14 +46,23 @@ class EndpointParameterIndex private constructor(model: Model) : KnowledgeIndex
             }
         }
 
+    /**
+     * Get the [operationContextParams](https://smithy.io/2.0/additional-specs/rules-engine/parameters.html#smithy-rules-operationcontextparams-trait)
+     * for an operation.
+     *
+     * @param op the operation shape to get context params for.
+     */
+    fun operationContextParams(op: OperationShape): MutableMap<String, OperationContextParamDefinition>? =
+        model.operationShapes.find { op.id == it.id }?.getTrait<OperationContextParamsTrait>()?.parameters
```

Comment:
**Question**: Why is looking up this operation shape in the model necessary? Can't you retrieve the trait directly off the input parameter?

```kotlin
fun operationContextParams(op: OperationShape) =
    op.getTrait<OperationContextParamsTrait>()?.parameters
```